histogram.js

```diff
@@ -72,27 +75,39 @@ class HistogramElement extends HTMLElement {
     this._histo = core({
       title: this.label,
     });
-    
     this.shadowRoot.appendChild(this._histo.el);
-
     const broker = getDataBroker(this);
+
+    const loader = this.shadowRoot.querySelector('.samplingLoader');
+
+    broker.onEvent('data-request-start', () => {
+      loader.style.display = 'block';
+      this._histo.update([]);
+    });
+
+    broker.onEvent('data-streaming-start', () => {
+        loader.style.display = 'none';
+    });
+
     if (broker) {
       let data = [];
       this._histo.update(data);
-
       broker.onEvent(this.brokerKey, (data) => {
-
-        var d = Object.keys(data).filter(function (i) {
-          return data[i] != "0"
-        }).map(function (k) {
-          return [+k, +data[k]]
-        });
-
-        if (this.ignoreOutliers) {
-          d = iobioviz.layout.outlier()(d);
+        if (loader.style.display === 'block') {
```

Comment:
I did the check there because there is a scenario where you abort a request and start a new one(e.g. check another chromosome). The old data is still streaming and updating the bar chart. The check here is for stopping update the bar chart while waiting for the new streaming data. 

You can reproduce the scenario I mentioned by deleting the check code. You will see that loading indicator will appear and the old bar chart is still updating.